### ISSUE TITLE
[one-line] False positive for `finally`

### ISSUE THREAD

**Issue by rhys-vdw** [palantir/tslint#3794]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.7.2
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
export function printUsabilityTest(): AsyncThunkAction {
  return async (dispatch, getState) => {
    try {
      // Open all navigation test sections.
      dispatch(expandAllSectionScreenshots())

      // Ensure that all screenshots have loaded successfully.
      const usabilityTest = getUsabilityTest(getState())
      const screenshots = usabilityTestScreenshotsSelector(getState(), usabilityTest)
      try {
        await Promise.all(screenshots.map((screenshot) =>
          dispatch(loadScreenshot(screenshot._clientId))
        ))
      } catch (error) {
        // Do nothing here. Just print the page without the failed image.
      }
      window.print()
    }
  }
}

```

with `tslint.json` configuration:

```json
{
  "extends": [
    "tslint:recommended",
    "tslint-eslint-rules",
    "tslint-react"
  ],
  "defaultSeverity": "error",
  "rules": {
    "align": [true, "elements", "members", "parameters", "statements"],
    "array-bracket-spacing": [true, "never"],
    "curly": [true, "ignore-same-line"],
    "indent": [true, "spaces", 2],
    "interface-name": false,
    "jsx-curly-spacing": [true, "never"],
    "jsx-no-multiline-js": false,
    "jsx-boolean-value": [true, "never"],
    "max-classes-per-file": false,
    "max-line-length": [true, 120],
    "member-access": [true, "no-public"],
    "member-ordering": [true, {
      "order": [
        "static-field",
        "static-method",
        "instance-field",
        "constructor",
        "instance-method"
      ]
    }],
    "no-console": [true, "warn", "error", "dir"],
    "no-constant-condition": true,
    "no-debugger": true,
    "no-duplicate-switch-case": true,
    "no-unused-variable": [true, "check-parameters", { "ignorePattern": "^_" }],
    "no-var-keyword": true,
    "no-var-requires": false,
    "no-switch-case-fall-through": true,
    "object-curly-spacing": [true, "always"],
    "object-literal-sort-keys": false,
    "ordered-imports": false,
    "prefer-const": true,
    "quotemark": [true, "double", "jsx-double", "avoid-escape"],
    "semicolon": [true, "never"],
    "trailing-comma": [true, {
      "multiline": {
        "objects": "always",
        "arrays": "always",
        "functions": "ignore",
        "typeLiterals": "always",
        "imports": "always",
        "exports": "always"
      },
      "singleline": "never"
    }],
    "variable-name": {
      "options": [
        "allow-leading-underscore",
        "ban-keywords",
        "check-format",
        "allow-pascal-case"
      ]
    }
  },
  "linterOptions": {
    "exclude": ["./node_modules/**/*"]
  }
}
```

#### Actual behavior

![image](https://user-images.githubusercontent.com/822105/37952257-e1aa2bf6-31ea-11e8-9e8b-311127f94b15.png)

#### Expected behavior

No error

**Comment by rhys-vdw** [comment]
lol, I'm silly
